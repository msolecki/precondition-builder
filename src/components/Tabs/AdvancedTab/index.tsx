import React from 'react'
import Paper from '@material-ui/core/Paper'
import {makeStyles} from '@material-ui/core/styles'
import {FormDataInterface} from '../../App'

const useStyles = makeStyles(({spacing}) => ({
    root: {
        padding: spacing(3, 2),
    },
}))

interface ActivatedProps {
    data: FormDataInterface;
}

const AdvancedTab: React.FC<ActivatedProps> = (props: ActivatedProps): React.ReactElement => {
    const classes = useStyles()

    return (
        <Paper className={classes.root}>
            {JSON.stringify(props.data)}
        </Paper>
    )
}

export default AdvancedTab
