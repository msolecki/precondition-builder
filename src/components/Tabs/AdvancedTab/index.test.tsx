import * as React from 'react'
import {ReactTestRenderer} from 'react-test-renderer'
import {createShallow} from '@material-ui/core/test-utils'

import AdvancedTab, {AdvancedTabProps} from './index'

const component = (props: AdvancedTabProps): AdvancedTab => <AdvancedTab {...props}/>

describe('<Activated />', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    test('empty snapshot', () => {
        const wrapper: ReactTestRenderer = shallow(component({data: {}}))

        expect(wrapper).toMatchSnapshot()
    })
})
