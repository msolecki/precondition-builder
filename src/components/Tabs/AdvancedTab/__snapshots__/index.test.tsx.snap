// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activated /> empty snapshot 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
>
  <JSONPretty
    data=""
    id="json-pretty"
    json={Object {}}
    silent={true}
    space={2}
    themeClassName="__json-pretty__"
  />
</WithStyles(ForwardRef(Paper))>
`;
